### ISSUE TITLE
Use sed to turn Bazel's C compiler into the C++ compiler by replacing an optional 'cc' followed by a version string with '++-<version>'.  This works for GCC/G++ and Clang/++. [phab:D305]

### ISSUE THREAD

**Issue by kythe** [google/kythe#525]
_Created by **shahms** at 2015-06-02 19:52:07_: 

Fix the C compiler to C++ compiler translation.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-06-02 19:56:38_:

On my OS X machine:
```
$ sed -e 's|\(cc\)\?\(-.*\)\?$|++\2|'
clang
clang
clang-3.5
clang-3.5
gcc
gcc
gcc-3.5
gcc-3.5
```
The sed command doesn't seem to be doing anything.

**Comment by kythe** [comment]
_**shahms** wrote at 2015-06-02 19:59:53_:

  - Fix sed command to use standard / delimiters and extended regular expressions.

**Comment by kythe** [comment]
_**shahms** wrote at 2015-06-02 20:01:07_:

>>! In (earlier comment), zarko wrote:
> On my OS X machine:
> ```
> $ sed -e 's|\(cc\)\?\(-.*\)\?$|++\2|'
> clang
> clang
> clang-3.5
> clang-3.5
> gcc
> gcc
> gcc-3.5
> gcc-3.5
> ```
> The sed command doesn't seem to be doing anything.

Yeah, I was using the GNU extension that allows using any delimiter for s commands.  Fixed (also, OSX supports -E to use extended rather than basic regular expressions).

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2015-06-02 20:45:54_

**Comment by kythe** [comment]
_**shahms** closed via commit 5abe4be6a18578dbf2af282b721fb3202b2823a0 at 2015-06-02 20:47:58_
